pkg/serializer/compression/internal/no_strategy.go

```diff
@@ -0,0 +1,63 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+// Package compression provides a set of functions for compressing with zlib / zstd
+package compression
+
+import (
+	"bytes"
+)
+
+// NoopStrategy is the strategy for when serializer_compressor_kind is neither zlib nor zstd
+type NoopStrategy struct {
+}
+
+// NewNoopStrategy returns a new NoopStrategy for when kind is neither zlib nor zstd
+func NewNoopStrategy() *NoopStrategy {
+	return &NoopStrategy{}
+}
+
+// Compress implements the Compress method for NoopStrategy to satisfy the Compressor interface
+func (s *NoopStrategy) Compress(src []byte) ([]byte, error) {
+	return src, nil
+}
+
+// Decompress implements the Decompress method for NoopStrategy to satisfy the Compressor interface
+func (s *NoopStrategy) Decompress(src []byte) ([]byte, error) {
+	return src, nil
+}
+
+// CompressBound implements the CompressBound method for NoopStrategy to satisfy the Compressor interface
+func (s *NoopStrategy) CompressBound(sourceLen int) int {
+	return sourceLen
+}
+
+// ContentEncoding implements the ContentEncoding method for NoopStrategy to satisfy the Compressor interface
+func (s *NoopStrategy) ContentEncoding() string {
+	return ""
+}
+
+// NoopZipper is the zipper for when the serializer_compressor_kind is neither zlib nor zstd
+type NoopZipper struct{}
+
+// Write implements the Write method for NoopZipper to satisfy the Zipper interface
+func (s NoopZipper) Write([]byte) (int, error) {
+	return 0, nil
```

Comment:
Should the no-op zipper pass along the unmodified bytes to the byte buffer?
Right now if the no-op gets selected it will just throw away the input.

Reply:
I am not sure that matters since we would only create the hypothetical no-op streamcompressor if `zlib` and `zstd` build tag are not included. But if that is the case, right now we just error https://github.com/DataDog/datadog-agent/blob/main/pkg/serializer/internal/stream/compressor_nozlib.go .  (`StreamCompressor` is created in `NewCompressor` func)

I'll update the file to account for the `zstd` build tag, good catch